pkg/plugins/connection/ssh.go

```diff
@@ -32,7 +32,13 @@ func (s SSH) String() string {
 
 // SSHOpen is a Connection plugin that opens a connection with a device
 type SSHOpen struct {
-	Meta *gornir.TaskMetadata // Task metadata
+	Meta           *gornir.TaskMetadata                                         // Task metadata
+	ClientConfigFn func(*gornir.Host, gornir.Logger) (*ssh.ClientConfig, error) // SSH client configuration
```

Comment:
maybe `func(*gornir.Host, gornir.Logger) (*ssh.ClientConfig, error)` should be a type of its own:

```
type ClientConfigFn func(*gornir.Host, gornir.Logger) (*ssh.ClientConfig, error)
```

so you can reference it in the struct:
```
 type SSHOpen struct {
    ClientConfigFn ClientConfigFn
```

Reply:
Sounds good!